docs/src/main/docs/health/02_health_in_k8s.adoc

```diff
@@ -0,0 +1,217 @@
+
+=== Readiness probe
+
+The readiness probe is used to avoid routing requests to the pod until it is
+ ready to accept traffic. When Kubernetes gives up on a readiness probe, the
+ pod is not restarted, it is removed from the service routing.
+
+NOTE: A bad readiness probe can result in a cascading effect where all pods are
+ removed from the service routing.
+
+We recommend the following:
+
+* Be conservative when checking shared dependencies
+* Be aggressive when checking local dependencies
+* Set `failureThreshold` according to `periodSeconds` in order to accomodate
+ temporary errors
+
+== Troubleshooting probes
+
+Failed probes are recorded as events associated with their corresponding pods.
+ The event message will only contain the status code information.
+
+[source,bash]
+.Get the events of a single pod
+----
+POD_NAME=$(kubectl get pod -l app=acme -o jsonpath='{.items[0].metadata.name}') # <1>
+kubectl get event --field-selector involvedObject.name=${POD_NAME} # <2>
+----
+<1> Get the effective pod name by filtering pods with the label `app=acme`
+<2> Filter the events for the pod
+
+TIP: Create log messages in your health check implementation when setting a
+ `DOWN` status, this will allow you to correlate the cause of a failed probe.
+
+== Example
+
+The example below illustrates the usages of the Helidon health API for an
+ application that implements health endpoints for both probes as well as a
+ Kubernetes service and deployment.
+The application code dissociates the health endpoints from the default routes
+ so that they are not exposed by the service.
+
+[source,java]
+.Application code
+----
+Routing healthRouting = Routing.builder()
+        .register(JsonSupport.create())
+        .register(HealthSupport.builder()
+                .webContext("/live") // <1>
+                .add(HealthChecks.healthChecks()) // <2>
+                .build())
+        .register(HealthSupport.builder()
+                .webContext("/ready") // <3>
+                .add(() -> HealthCheckResponse.named("database").up().build()) // <4>
+                .build())
+        .build();
+
+Routing defaultRouting = Routing.builder()
+        .any((req, res) -> res.send("It works!")) // <5>
+        .build();
+
+WebServer server = WebServer.builder(defaultRouting)
+        .config(ServerConfiguration.builder()
+                .port(8080) // <6>
+                .addSocket("health", SocketConfiguration.builder() // <7>
+                        .port(8081)
+                        .build())
+                .build())
+        .addNamedRouting("health", healthRouting) // <8>
+        .build();
+
+server.start();
+----
+<1> The health service for the `liveness` probe is exposed at `/live`
+<2> Using the built-in health checks for the `liveness` probe
+<3> The health service for the `readiness` probe is exposed at `/ready`
+<4> Using a custom health check for a pseudo database that is always `UP`
+<5> The default routes, returns `It works!` for any request
```

Comment:
"routes" or "route" ?

The default route: returns `It works!` for any request.